src/mgr/DaemonServer.cc

```diff
@@ -385,6 +390,7 @@ bool DaemonServer::handle_report(MMgrReport *m)
     return true;
   }
 
+  // Look up the DaemonState
```

Comment:
shall we guard the access to `daemon_state` with `lock` here also?

Reply:
It's not necessary until we are doing some updates (in the "Update the DaemonState" block below).

In the "constructing new DaemonState" path we are making updates before inserting it into the index, so nobody else can see it yet.